includes/compat/class-customer.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Backwards Compatibility Handler for Customers.
+ *
+ * @package     EDD
+ * @subpackage  Compat
+ * @copyright   Copyright (c) 2018, Easy Digital Downloads, LLC
+ * @license     http://opensource.org/licenses/gpl-2.0.php GNU Public License
+ * @since       3.0.0
+ */
+namespace EDD\Compat;
+
+// Exit if accessed directly
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Customer Class.
+ *
+ * @since 3.0.0
+ */
+class Customer extends Base {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 3.0.0
+	 */
+	public function __construct() {
+		$this->set_component( 'customer' );
+		$this->hooks();
+	}
+
+	/**
+	 * Magic method to handle calls to method that no longer exist.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param string $name      Name of the method.
+	 * @param array  $arguments Enumerated array containing the parameters passed to the $name'ed method.
+	 * @return mixed Dependent on the method being dispatched to.
+	 */
+	public function __call( $name, $arguments ) {
+		switch ( $name ) {
+			case 'add':
+			case 'insert':
+				return edd_add_customer( $arguments[0] );
+				break;
+			case 'update':
+				return edd_update_customer( $arguments[0], $arguments[1] );
```

Comment:
I think we should be careful about making assumptions that arguments will be defined here and some of the other cases. Both arguments for `edd_update_customer()` are technically optional, for instance.